test/new-e2e/tests/agent-subcommands/health/health_nix_test.go

```diff
@@ -39,10 +40,12 @@ func (v *linuxHealthSuite) TestDefaultInstallUnhealthy() {
 	))
 
 	// agent should be unhealthy because the key is invalid
-	_, err := v.Env().Agent.Client.Health()
-	if err == nil {
-		assert.Fail(v.T(), "agent expected to be unhealthy, but no error found!")
-		return
-	}
-	assert.Contains(v.T(), err.Error(), "Agent health: FAIL")
+	assert.EventuallyWithT(v.T(), func(t *assert.CollectT) {
+		_, err := v.Env().Agent.Client.Health()
+		if err == nil {
+			assert.Fail(t, "agent expected to be unhealthy, but no error found!")
+			return
+		}
+		assert.Contains(t, err.Error(), "Agent health: FAIL")
```

Comment:
There is an [ErrorContains](https://pkg.go.dev/github.com/stretchr/testify/assert#ErrorContains) helper that might be useful here ?
```suggestion
		assert.ErrorContains(t, err, "Agent health: FAIL")
```